### PR TITLE
Add Pipeline class

### LINKED ISSUE #64: Create class Pipeline
### 🚀 Feature Request

Class Pipeline should:
- combine model and features
- have fit and forecast methods.
- implement logic for feature generation
- apply inverse_transform to answers

### Motivation

This structure will help make prediction easier with less boilerplate code.

### Proposal

Create class Pipeline.
As arguments Pipeline expects `model`, `features`, `horizon`.
Implement 2 methods: fit and forecast.

.fit:
- expects TSDataset
- fit and apply features on given TSDataset
- fit model

.forecast:
- call make_future method with horizon parameter
- apply model to future dataset
- call inverse_transform method on forecast dataset
- return forecast dataset

### Test cases

1. Features fitted
2. Model fitted 
3. .forecast method returns same result as script without it

### Alternatives

_No response_

### Additional context

_No response_

### Checklist

- [X] Added feature request
- [X] Added motivation
- [X] Added proposal

### PR DESCRIPTION
**Pull request by alex-hse-repository**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

## Before submitting (checklist)

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna-ts/blob/master/CONTRIBUTING.md)?
- [x] Did you check the code style? `make lint` (`poetry install -E style`).
- [x] Did you make sure to update the docs? We use Numpy format for all the methods and classes.
- [x] Did you write any new necessary tests?
- [x] Did you check that your code passes the unit tests `pytest tests/` ? 
- [ ] Did you add your new functionality to the docs?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna-ts/blob/master/CHANGELOG.md)?

## Type of Change

- [ ] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Proposed Changes

  -Add Pipeline class

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Closing issues
closes #64 
Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: etna/pipeline/pipeline.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        Parameters
+        ----------
+        ts:
+            Dataset with timeseries data
+        Returns
+        -------
+        self :
+            Fitted Pipeline instance
+        """
+        ts = deepcopy(ts)
```
Comment (Mr-Geekman) [inline etna/pipeline/pipeline.py:43]: May be we can use `ts.copy(deep=True)` in this case.

File: etna/pipeline/pipeline.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+from etna.datasets import TSDataset
+from etna.models.base import Model
+from etna.transforms.base import Transform
+
+
+class Pipeline:
+    """Pipeline of transforms with a final estimator."""
+
+    def __init__(self, model: Model, transforms: Iterable[Transform] = [], horizon: int = 1):
```
Comment (iKintosh) [inline etna/pipeline/pipeline.py:12]: Should not use mutable objects as defaults 
```suggestion
    def __init__(self, model: Model, transforms: Iterable[Transform] = (), horizon: int = 1):
```

File: etna/pipeline/pipeline.py (hunk 1 above)
Comment (iKintosh) [reply etna/pipeline/pipeline.py:43]: TSDataset does not have method copy

**Review by iKintosh — CHANGES_REQUESTED** [review]
Looks good except for the list as default argument

File: etna/pipeline/pipeline.py (hunk 1 above)
Comment (iKintosh) [reply etna/pipeline/pipeline.py:43]: In this case we do not need deep copy because we work with TSDataset (has `raw_df`) and not with pd.DataFrame

**Review by iKintosh — APPROVED** [review]
🔥

**Pull request merged**